ReactWindows/ReactNative.Shared/UIManager/UIManagerModule.cs

```diff
@@ -511,6 +516,39 @@ public void configureNextLayoutAnimation(JObject config, ICallback success, ICal
 
         #endregion
 
+        #region Sync React Methods
+
+        /// <summary>
+        /// Gets the constants for the view manager.
+        /// </summary>
+        /// <param name="viewManagerName">The view manager name.</param>
+        /// <returns>The view manager constants.</returns>
+        [ReactMethod(IsBlockingSynchronousMethod = true)]
+        public IDictionary<string, object> getConstantsForViewManager(string viewManagerName)
+        {
+            var viewManager = _uiImplementation.ResolveViewManager(viewManagerName);
+
+            using (Tracer.Trace(Tracer.TRACE_TAG_REACT_BRIDGE, "UIManagerModule.getConstantsForViewManager")
+                .With("ViewManager", viewManagerName)
+                .With("Lazy", true)
+                .Start())
+            {
+                return CreateConstantsForViewManager(viewManager, null, null, null, _customDirectEvents);
```

Comment:
>_customDirectEvents [](start = 84, length = 19)

Threading concern: this dictionary has to be changed into a Concurrent one, it is modified by JS thread and accessed by layout manager one.

Reply:
Good catch, I think it's actually the main UI thread and the JS thread. It's read on the UI thread for the native animated module and written to on the JS thread. I think Android has the same bug.

_Sigh_ - UIManager now has 3 different types of thread access :-(

---
In reply to: [174852891](https://github.com/Microsoft/react-native-windows/pull/1717#discussion_r174852891) [](ancestors = 174852891)

Reply:
We should probably fix on Android :-|

---
In reply to: [174886730](https://github.com/Microsoft/react-native-windows/pull/1717#discussion_r174886730) [](ancestors = 174886730,174883624,174852891)

Reply:
This is going to be ugly - this is a triple nested dictionary. Rather than going concurrent all the way down, I'm going to do a bit of locking :-/

---
In reply to: [174980315](https://github.com/Microsoft/react-native-windows/pull/1717#discussion_r174980315) [](ancestors = 174980315,174886730,174883624,174852891)